src/util/s3select.js

```diff
@@ -99,8 +135,41 @@ class S3SelectStream extends Transform {
         }
     }
 
+    async send_stats() {
+        const xml_buff = Buffer.from(xml_utils.encode_xml(this.stats));
+        const total_length_bytes =
+            xml_buff.length +
+            S3SelectStream.stats_message_headers.length +
+            S3SelectStream.prelude_length_bytes +
+            S3SelectStream.crc_length_bytes;
+
+        const buffer = Buffer.alloc(S3SelectStream.prelude_length_bytes);
+        let index = 0;
+        let message_crc = 0;
+
+        //prelude:
+        index = buffer.writeUInt32BE(total_length_bytes);
+        index = buffer.writeUInt32BE(S3SelectStream.stats_message_headers.length, index);
+        buffer.writeInt32BE(crc32.buf(buffer.subarray(0, index)), index);
+        this.push(buffer);
+        message_crc = crc32.buf(buffer);
+        //headers:
+        this.push(S3SelectStream.stats_message_headers);
+        message_crc = crc32.buf(S3SelectStream.stats_message_headers, message_crc);
+        //body:
+        this.push(xml_buff);
+        message_crc = crc32.buf(xml_buff, message_crc);
+        //message crc:
+        const buffer2 = Buffer.alloc(S3SelectStream.crc_length_bytes);
+        buffer2.writeInt32BE(message_crc);
+        this.push(buffer2);
+    }
+
     async _transform(chunk, encoding, cb) {
         try {
+            //TODO - consider compression?
```

Comment:
In case you are not planning to consider compression in the near future, I think it will be better to add a comment that those calculations do not consider compression...